lms/djangoapps/django_comment_client/forum/views.py

```diff
@@ -145,17 +145,26 @@ def get_threads(request, course, discussion_id=None, per_page=THREADS_PER_PAGE):
     #if the user requested a group explicitly, give them that group, otherwise, if mod, show all, else if student, use cohort
 
     group_id = request.GET.get('group_id')
+    is_cohorted = is_commentable_cohorted(course.id, discussion_id)
 
-    if group_id == "all":
+    if group_id in ("all", "None"):
         group_id = None
 
-    if not group_id:
-        if not has_permission(request.user, "see_all_cohorts", course.id):
-            group_id = get_cohort_id(request.user, course.id)
-            if not group_id and get_cohorted_threads_privacy(course.id) == 'cohort-only':
-                default_query_params['exclude_groups'] = True
+
+    if not has_permission(request.user, "see_all_cohorts", course.id):
+        group_id = get_cohort_id(request.user, course.id)
+        if not group_id and get_cohorted_threads_privacy(course.id) == 'cohort-only':
+            default_query_params['exclude_groups'] = True
```

Comment:
Nit: I think this code would be much easier to follow if written like:

```python
if has_permission(request.user, "see_all_cohorts", course.id):
    group_id = request.GET.get('group_id')
    if group_id in ("all", "None"):
        group_id = None
else:
    group_id = get_cohort_id(request.user, course.id)
    if not group_id and get_cohorted_threads_privacy(course.id) == 'cohort-only':
        default_query_params['exclude_groups'] = True
```

Also, it looks to me like `?group_id=all` and `?group_id=None` are treated exactly the same. Is that right?